x-pack/plugins/ml/public/application/services/http_service.ts

```diff
@@ -102,3 +102,105 @@ export function fromHttpHandler<T>(input: string, init?: RequestInit): Observabl
     };
   });
 }
+
+/**
+ * ML Http Service
+ */
+export class HttpService {
```

Comment:
To remove the duplicate code here, could `http` and `$http` be changed to use an instance of this class which is a singleton initialised using `getHttp`?

Reply:
Actually, ignore this suggestion. I don't know what side effects it would have.